src/PrestaShopBundle/Bridge/Smarty/BreadcrumbsAndTitleHydrator.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShopBundle\Bridge\Smarty;
+
+use Configuration;
+use Link;
+use PrestaShop\PrestaShop\Adapter\LegacyContext;
+use PrestaShopBundle\Bridge\Controller\ControllerConfiguration;
+use Tab;
+use Tools;
+
+/**
+ * Class BreadcrumbsAndTitleHydrator hydrate breadcrumbs in ControllerConfiguration
```

Comment:
We should explain the goal of hydrator class a bit more, since "hydrate" can be interpreted as different ways

Maybe just a short description like `it prepares all the necessary variables for smarty templates based on controller configuration` would do (in all classes since we can't comment the namespace itself 🤷‍♂️ )